[qt5-declarative] x64-linux build error caused by jasper 4.0.0 update

**Host Environment**

- OS: Ubuntu 22.04
- Compiler: gcc 11.3.0

**To Reproduce**

Steps to reproduce the behavior:

`sudo apt install libheif-dev`
`./vcpkg install jasper`
`./vcpkg install qt5-declarative`

**Failure logs**

Relevant content from `vcpkg/buildtrees/qt5-declarative/package-build-x64-linux-dbg-err.log`

```
/usr/bin/ld: /home/miriam/vcpkg/installed/x64-linux/debug/lib/libjasperd.a(heic_dec.c.o): in function `jas_heic_decode':
/home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:163: undefined reference to `heif_get_version'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:163: undefined reference to `heif_get_version_number'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:167: undefined reference to `heif_get_version_number'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:169: undefined reference to `heif_get_version'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:187: undefined reference to `heif_context_alloc'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:193: undefined reference to `heif_context_read_from_memory_without_copy'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:196: undefined reference to `heif_context_get_primary_image_handle'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:198: undefined reference to `heif_image_handle_get_width'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:199: undefined reference to `heif_image_handle_get_height'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:222: undefined reference to `heif_decode_image'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:229: undefined reference to `heif_image_get_colorspace'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:230: undefined reference to `heif_image_get_chroma_format'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:235: undefined reference to `heif_image_get_plane_readonly'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:302: undefined reference to `heif_image_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:304: undefined reference to `heif_context_free'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:306: undefined reference to `heif_image_handle_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:316: undefined reference to `heif_image_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:319: undefined reference to `heif_context_free'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:322: undefined reference to `heif_image_handle_release'
/usr/bin/ld: /home/miriam/vcpkg/installed/x64-linux/debug/lib/libjasperd.a(heic_enc.c.o): in function `jas_heic_encode':
/home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:211: undefined reference to `heif_get_version'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:211: undefined reference to `heif_get_version_number'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:215: undefined reference to `heif_get_version_number'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:217: undefined reference to `heif_get_version'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:232: undefined reference to `heif_context_alloc'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:237: undefined reference to `heif_context_get_encoder_for_format'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:245: undefined reference to `heif_encoder_set_lossless'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:247: undefined reference to `heif_encoder_set_lossy_quality'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:263: undefined reference to `heif_image_create'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:278: undefined reference to `heif_image_add_plane'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:285: undefined reference to `heif_image_get_plane'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:294: undefined reference to `heif_context_encode_image'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:303: undefined reference to `heif_context_write'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:309: undefined reference to `heif_image_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:312: undefined reference to `heif_encoder_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:315: undefined reference to `heif_context_free'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:327: undefined reference to `heif_image_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:330: undefined reference to `heif_encoder_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:333: undefined reference to `heif_context_free'
/usr/bin/ld: /home/miriam/vcpkg/installed/x64-linux/debug/lib/libjasperd.a(heic_dec.c.o): in function `jas_heic_decode':
/home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:163: undefined reference to `heif_get_version'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:163: undefined reference to `heif_get_version_number'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:167: undefined reference to `heif_get_version_number'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:169: undefined reference to `heif_get_version'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:187: undefined reference to `heif_context_alloc'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:193: undefined reference to `heif_context_read_from_memory_without_copy'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:196: undefined reference to `heif_context_get_primary_image_handle'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:198: undefined reference to `heif_image_handle_get_width'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:199: undefined reference to `heif_image_handle_get_height'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:222: undefined reference to `heif_decode_image'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:229: undefined reference to `heif_image_get_colorspace'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:230: undefined reference to `heif_image_get_chroma_format'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:235: undefined reference to `heif_image_get_plane_readonly'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:302: undefined reference to `heif_image_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:304: undefined reference to `heif_context_free'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:306: undefined reference to `heif_image_handle_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:316: undefined reference to `heif_image_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:319: undefined reference to `heif_context_free'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:322: undefined reference to `heif_image_handle_release'
/usr/bin/ld: /home/miriam/vcpkg/installed/x64-linux/debug/lib/libjasperd.a(heic_enc.c.o): in function `jas_heic_encode':
/home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:211: undefined reference to `heif_get_version'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:211: undefined reference to `heif_get_version_number'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:215: undefined reference to `heif_get_version_number'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:217: undefined reference to `heif_get_version'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:232: undefined reference to `heif_context_alloc'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:237: undefined reference to `heif_context_get_encoder_for_format'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:245: undefined reference to `heif_encoder_set_lossless'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:247: undefined reference to `heif_encoder_set_lossy_quality'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:263: undefined reference to `heif_image_create'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:278: undefined reference to `heif_image_add_plane'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:285: undefined reference to `heif_image_get_plane'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:294: undefined reference to `heif_context_encode_image'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:303: undefined reference to `heif_context_write'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:309: undefined reference to `heif_image_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:312: undefined reference to `heif_encoder_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:315: undefined reference to `heif_context_free'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:327: undefined reference to `heif_image_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:330: undefined reference to `heif_encoder_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:333: undefined reference to `heif_context_free'
/usr/bin/ld: /home/miriam/vcpkg/installed/x64-linux/debug/lib/libjasperd.a(heic_dec.c.o): in function `jas_heic_decode':
/home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:163: undefined reference to `heif_get_version'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:163: undefined reference to `heif_get_version_number'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:167: undefined reference to `heif_get_version_number'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:169: undefined reference to `heif_get_version'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:187: undefined reference to `heif_context_alloc'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:193: undefined reference to `heif_context_read_from_memory_without_copy'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:196: undefined reference to `heif_context_get_primary_image_handle'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:198: undefined reference to `heif_image_handle_get_width'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:199: undefined reference to `heif_image_handle_get_height'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:222: undefined reference to `heif_decode_image'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:229: undefined reference to `heif_image_get_colorspace'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:230: undefined reference to `heif_image_get_chroma_format'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:235: undefined reference to `heif_image_get_plane_readonly'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:302: undefined reference to `heif_image_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:304: undefined reference to `heif_context_free'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:306: undefined reference to `heif_image_handle_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:316: undefined reference to `heif_image_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:319: undefined reference to `heif_context_free'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:322: undefined reference to `heif_image_handle_release'
/usr/bin/ld: /home/miriam/vcpkg/installed/x64-linux/debug/lib/libjasperd.a(heic_enc.c.o): in function `jas_heic_encode':
/home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:211: undefined reference to `heif_get_version'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:211: undefined reference to `heif_get_version_number'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:215: undefined reference to `heif_get_version_number'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:217: undefined reference to `heif_get_version'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:232: undefined reference to `heif_context_alloc'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:237: undefined reference to `heif_context_get_encoder_for_format'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:245: undefined reference to `heif_encoder_set_lossless'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:247: undefined reference to `heif_encoder_set_lossy_quality'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:263: undefined reference to `heif_image_create'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:278: undefined reference to `heif_image_add_plane'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:285: undefined reference to `heif_image_get_plane'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:294: undefined reference to `heif_context_encode_image'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:303: undefined reference to `heif_context_write'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:309: undefined reference to `heif_image_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:312: undefined reference to `heif_encoder_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:315: undefined reference to `heif_context_free'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:327: undefined reference to `heif_image_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:330: undefined reference to `heif_encoder_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:333: undefined reference to `heif_context_free'
/usr/bin/ld: /home/miriam/vcpkg/installed/x64-linux/debug/lib/libjasperd.a(heic_dec.c.o): in function `jas_heic_decode':
/home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:163: undefined reference to `heif_get_version'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:163: undefined reference to `heif_get_version_number'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:167: undefined reference to `heif_get_version_number'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:169: undefined reference to `heif_get_version'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:187: undefined reference to `heif_context_alloc'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:193: undefined reference to `heif_context_read_from_memory_without_copy'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:196: undefined reference to `heif_context_get_primary_image_handle'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:198: undefined reference to `heif_image_handle_get_width'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:199: undefined reference to `heif_image_handle_get_height'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:222: undefined reference to `heif_decode_image'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:229: undefined reference to `heif_image_get_colorspace'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:230: undefined reference to `heif_image_get_chroma_format'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:235: undefined reference to `heif_image_get_plane_readonly'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:302: undefined reference to `heif_image_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:304: undefined reference to `heif_context_free'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:306: undefined reference to `heif_image_handle_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:316: undefined reference to `heif_image_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:319: undefined reference to `heif_context_free'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_dec.c:322: undefined reference to `heif_image_handle_release'
/usr/bin/ld: /home/miriam/vcpkg/installed/x64-linux/debug/lib/libjasperd.a(heic_enc.c.o): in function `jas_heic_encode':
/home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:211: undefined reference to `heif_get_version'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:211: undefined reference to `heif_get_version_number'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:215: undefined reference to `heif_get_version_number'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:217: undefined reference to `heif_get_version'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:232: undefined reference to `heif_context_alloc'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:237: undefined reference to `heif_context_get_encoder_for_format'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:245: undefined reference to `heif_encoder_set_lossless'
usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:247: undefined reference to `heif_encoder_set_lossy_quality'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:263: undefined reference to `heif_image_create'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:278: undefined reference to `heif_image_add_plane'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:285: undefined reference to `heif_image_get_plane'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:294: undefined reference to `heif_context_encode_image'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:303: undefined reference to `heif_context_write'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:309: undefined reference to `heif_image_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:312: undefined reference to `heif_encoder_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:315: undefined reference to `heif_context_free'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:327: undefined reference to `heif_image_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:330: undefined reference to `heif_encoder_release'
/usr/bin/ld: /home/miriam/vcpkg/buildtrees/jasper/src/5eb260a2e4-fc3d85658e.clean/src/libjasper/heic/heic_enc.c:333: undefined reference to `heif_context_free'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:81: ../../bin/qmlpreview] Error 1
make[1]: *** [Makefile:231: sub-qmlpreview-make_first] Error 2
make[1]: *** Waiting for unfinished jobs....
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:88: ../../bin/qmlprofiler] Error 1
make[1]: *** [Makefile:206: sub-qmlprofiler-make_first] Error 2
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:70: ../../bin/qmltestrunner] Error 1
make[1]: *** [Makefile:281: sub-qmltestrunner-make_first] Error 2
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:78: ../../bin/qml] Error 1
make[1]: *** [Makefile:181: sub-qml-make_first] Error 2
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:85: ../../bin/qmleasing] Error 1
make[1]: *** [Makefile:256: sub-qmleasing-make_first] Error 2
make: *** [Makefile:75: sub-tools-make_first] Error 2
```

This error is caused by https://github.com/microsoft/vcpkg/pull/28244, qt5-declarative installs successfully if I check out to commit older before the merge of https://github.com/microsoft/vcpkg/pull/28244.

Attempted the below and it still fails,
`sudo apt remove libheif-dev`
`./vcpkg install libheif`
`./vcpkg install jasper`
`./vcpkg install qt5-declarative`

Attempted the below though I need libheif for other purposes but it still fails,
`sudo apt remove libheif-dev` 
`./vcpkg install jasper`
`./vcpkg install qt5-declarative`

Can we please fix this?

--- a/ports/jasper/portfile.cmake
+++ b/ports/jasper/portfile.cmake
@@ -23,8 +23,9 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
+        ${OUT_FEATURE_OPTIONS}
+        -DJAS_ENABLE_LIBHEIF=OFF # found via find_library instead of find_package
         -DJAS_ENABLE_LIBJPEG=ON
-        -DJAS_ENABLE_OPENGL=${JAS_ENABLE_OPENGL}
         -DJAS_ENABLE_DOC=OFF
         -DJAS_ENABLE_LATEX=OFF
         -DJAS_ENABLE_PROGRAMS=OFF

--- a/ports/jasper/portfile.cmake
+++ b/ports/jasper/portfile.cmake
@@ -23,7 +23,7 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
-        ${OUT_FEATURE_OPTIONS}
+        ${FEATURE_OPTIONS}
         -DJAS_ENABLE_LIBHEIF=OFF # found via find_library instead of find_package
         -DJAS_ENABLE_LIBJPEG=ON
         -DJAS_ENABLE_DOC=OFF

--- a/ports/jasper/vcpkg.json
+++ b/ports/jasper/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "jasper",
   "version": "4.0.0",
+  "port-version": 1,
   "description": "Open source implementation of the JPEG-2000 Part-1 standard",
   "homepage": "https://github.com/jasper-software/jasper",
   "license": null,

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3322,7 +3322,7 @@
     },
     "jasper": {
       "baseline": "4.0.0",
-      "port-version": 0
+      "port-version": 1
     },
     "jbig2dec": {
       "baseline": "0.19",

--- a/versions/j-/jasper.json
+++ b/versions/j-/jasper.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "dbc2d3298015b443afd029caa32dd107071ef810",
+      "version": "4.0.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "93774ef7923c7c1c73c30011afe0105530e71545",
       "version": "4.0.0",
       "port-version": 0

--- a/versions/j-/jasper.json
+++ b/versions/j-/jasper.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "dbc2d3298015b443afd029caa32dd107071ef810",
+      "git-tree": "1da5fa9a12bf207bf641f903856418e3aed7258d",
       "version": "4.0.0",
       "port-version": 1
     },

